### ISSUE TITLE
2022 Jupyter Widgets Dev Meetings

### ISSUE THREAD

**Issue by jasongrout** [jupyter-widgets/team-compass#13]
Here is a record of the [agenda/minutes](https://hackmd.io/5XWHyOoLTRqyXzEHsVmxXg) for our [weekly meeting](https://github.com/jupyter-widgets/team-compass#weekly-team-meetings). Everyone is welcome to join.

Let's avoid using this thread for discussion. If you'd like to discuss something in the minutes, open a separate issue and reference this thread.

## Meeting Notes

* [20 September 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1259305971)
* [13 September 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1259305669)
* [06 September 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1259305088)
* [30 August 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1259304471)
* [23 August 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1226928369)
* [16 August 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1224327178)
* [9 August 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1213164679)
* [2 August 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204873430)
* [26 July 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204873024)
* [19 July 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204872686)
* [12 July 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204872216)
* [5 July 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204872001)
* [28 June 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204871850)
* [21 June 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204871653)
* [14 June 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204871317)
* [7 June 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204870716)
* [31 May 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204870547)
* [24 May 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204870105)
* [17 May 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204869714)
* [10 May 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204869398)
* [03 May 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204869222)
* [26 April 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204869075)
* [19 April 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204868771)
* [12 April 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204868390)
* [5 April 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204868131)
* [29 March 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204867905)
* [15 March 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204867783)
* [8 March 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1204867618)
* [1 March 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1056759181)
* [22 Feb 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1056758959)
* [15 Feb 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1056758750)
* [08 Feb 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1033587717)
* [01 Feb 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1033587027)
* [25 Jan 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1021738253)
* [18 Jan 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1020572957)
* [11 Jan 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1020572807)
* [04 Jan 2022](https://github.com/jupyter-widgets/team-compass/issues/13#issuecomment-1020572651)

**Comment by jasongrout** [comment]
## 4 Jan 2022
Attendees: @vidartf, @martinRenou, @ibdafna

* https://github.com/jupyter-widgets/ipywidgets/pull/3326

* Should we consider this for 8.0.0? https://github.com/jupyter-widgets/ipywidgets/pull/3335
  - @jasongrout: I would love to see this in 8.0

* This week:
    * Add Maarten's PR to the change log
    * Release RC0
    * Try to migrate some of our widget libraries
    * Document any changes pertaining to migrating widget packages from 7.x to 8.x
    * Document Maarten's changes to the protocol spec
    * Release RC1
    * Announce RC

**Comment by jasongrout** [comment]
## 11 Jan 2022

Attendees: @vidartf, @martinRenou, @SylvainCorlay, @jasongrout, @ibdafna

- Maarten: https://github.com/jupyter-widgets/ipywidgets/pull/3343
  - Bump protocol to 3.0, since there are communication channels set up in both directions. Having a browser that understand 2.0 talking to a kernel speaking 3.0 should lead to sliders jumping around (since the echo field is ignored). Having a browser that understands 3 talking to a kernel that understands 2 means that channels that are opened by the browser will be rejected since the major version number does not match in the comm_open of the kernel.


- Martin: Updated https://github.com/jupyter-widgets/ipywidgets/pull/3335
  - Needs review
  - Martin will check if there is some heuristic in comm info message to determine if comm target exists or not
  - We should make an update to the docs/ipykernel to have the comm info request have an error status if the comm target does not exist.
- Bug fix with nbconvert https://github.com/jupyter-widgets/ipywidgets/pull/3342
  - Looks good, basically adds the lab variables as a default for html manager if they are not alreadyd defined on the page
  - Could check the generated code to see if there is a difference with how the css is put on the page in the conditional
  -

**Comment by jasongrout** [comment]
## 18 Jan 2022

Attendees: @vidartf, @martinRenou, @SylvainCorlay, @jasongrout, @ibdafna

- Decision-making body: look for an email
- [Dependabot config](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates)
- https://github.com/jupyter-widgets/ipywidgets/pull/3343
- Needs review https://github.com/jupyter-widgets/ipywidgets/pull/3335
- HTML manager vs JupyterLab manager

**Comment by jasongrout** [comment]
## 25 Jan 2022

Attendees: @vidartf, @martinRenou, @ivanov, @jasongrout, @SylvainCorlay, @ibdafna, @maartenbreddels 

- Decision-making body
- Should we merge? https://github.com/jupyter-widgets/ipywidgets/pull/3335
  - Comm message protocol discussion: how do we determine if a comm target is registered?
    - A target can be registered, comms created, then the target unregistered
    - {target_name: [list of comms]}, and also return active comm targets, filtered by the target_name.
    - Jason will file a JEP proposing two changes: (Update: filed at https://github.com/jupyter/enhancement-proposals/issues/86)
- Porting 7.x widgets to 8.x
  - Good test cases:
    - [X] bqplot: uses phosphor https://github.com/bqplot/bqplot/pull/1404 works fine!
    - [ ] pythreejs: doesn't use es6 classes yet
    - https://github.com/mariobuikhuizen/ipyvue/pull/55/files
- Consider changing the meeting time for this group?
  - Perhaps after the decision-making body is formed?

**Comment by jtpio** [comment]
## 1 Feb 2022

Attendees: @jasongrout, @vidartf, @afonit, @martinRenou, @ibdafna

- Introductions

- https://github.com/jupyter-widgets/ipywidgets/issues/3357
- https://github.com/jupyter-widgets/ipywidgets/issues/3359
- https://github.com/jupyter-widgets/ipywidgets/pull/3360
- https://github.com/jupyter-widgets/ipywidgets/issues/3364
- https://github.com/jupyter-widgets/ipywidgets/pull/3365
- https://github.com/jupyter-widgets/ipywidgets/pull/3335


- Please comment on https://github.com/jupyter/enhancement-proposals/issues/86

- Close to RC. To release RC, resolve:
  - https://github.com/jupyter-widgets/ipywidgets/issues/3357
  - https://github.com/jupyter-widgets/ipywidgets/pull/3365
  - https://github.com/jupyter-widgets/ipywidgets/issues/3350
  - https://github.com/jupyter-widgets/ipywidgets/pull/3335


Grids on ipywidgets
- qgrid seems dead, uses slickgrid
- ipyaggrid uses aggrid, also not maintained
- recommend ipydatagrid
- See also https://github.com/glue-viz/glue-jupyter/pull/129

Scipy
- Deadline 11 Feb. Tutorial spearheaded by Matt
- Maarten is writing a react-like layer on top of ipywidgets, may submit a talk

**Comment by jtpio** [comment]
## 8 Feb 2022

Attendees: @williamstein, @vidartf, @martinRenou, @jtpio, @jasongrout, @ibdafna, @ianhi, @maartenbreddels 

Other things that we would need/want for 8.0?
- https://github.com/jupyter-widgets/ipywidgets/pull/3004

PRs for review:
- https://github.com/jupyter-widgets/ipywidgets/pull/3366
- https://github.com/jupyter-widgets/ipywidgets/pull/3367

Merged PRs:
- https://github.com/jupyter-widgets/ipywidgets/pull/3368
- https://github.com/jupyter-widgets/ipywidgets/pull/3373
- https://github.com/jupyter-widgets/ipywidgets/pull/3338
- https://github.com/jupyter-widgets/ipywidgets/pull/3335
- https://github.com/jupyter-widgets/ipywidgets/pull/3370

7.x release
- Martin backporting the control channel
- 7.x minor release later this week

8.0 RC release
- This week after two PRs above merged.

Discussion
- Is there a project that involves building a UI with drag and drop and widgets?
  - Cocalc collaborative whiteboard
  - https://github.com/voila-dashboards/voila-gridstack
  - Databricks has something, but not ipywidgets
  - Internal product (closed source) has been developed
  - [ipyflex](https://github.com/trungleduc/ipyflex)
  - https://gitlab.com/cosapp/cosapp_lab 

- Scipy talks
  - Tutorial going in
  - Maarten talking about building larger ipywidgets applications
  - Jason may submit a talk on core ipywidgets 8 (or maybe a lightning talk, depending on how much stuff there is)
  - Itay may submit a talk about ipydatagrid

- ipywidgets controlling scientific instruments
  - ESRF
  - 


- Moving state to the server side
  - https://github.com/jupyterlab/jupyterlab/pull/11599

### Naive Community Questions (answers above):

Question (@williamstein): is there any existing software that let you do things like insert sliders, buttons, etc., graphically on a canvas, then generate code and uses ipywidgets to implement things under the hood?  (Instead of laying everything out via code.)  I'm stumbling into implementing something like this, and want to compare notes.

@martinRenou: ~~Do you mean rendering widgets on a canvas? 
I remember discussing this with Wolf Vollprecht (having multiple renderers for ipywidgets: DOM, Canvas, Qt etc) this sounds hard to implement to me, but it could be possible?~~ (misunderstood the question)

**Comment by martinRenou** [comment]
## 15 Feb 2022
Attendees: @williamstein, @vidartf, @ibdafna, @ivanov 

Merged PRs:
 - https://github.com/jupyter-widgets/ipywidgets/pull/3380
 - https://github.com/jupyter-widgets/ipywidgets/pull/3377
 - https://github.com/jupyter-widgets/ipywidgets/pull/3337

Quick demo:
  - @williamstein can show you ipywidgets in his work-in-progress (very buggy!) [collaborative whiteboard](https://github.com/sagemathinc/cocalc/pull/5674).  (Must be before 10am.)

Releasing 8.0
- @ibdafna to cut the release

**Comment by martinRenou** [comment]
## 22 Feb 2022
Attendees: @vidartf, @jasongrout, @sylvaincorlay

Merged PRs:
- https://github.com/jupyter-widgets/ipywidgets/pull/2975
- https://github.com/jupyter-widgets/ipywidgets/pull/3389
- https://github.com/jupyter-widgets/ipywidgets/pull/3387

Discussion points for 8.0:
- https://github.com/jupyter-widgets/ipywidgets/issues/3392
  - Jason will work on this this week

**Comment by martinRenou** [comment]
## 1 March 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
|Gayle Ollington|NumFOCUS|@gollington |
| Vidar T Fauske| JP Morgan Chase | @vidartf |
| Steve Purves | Curvenote | @stevejpurves |

Merged PRs:
- https://github.com/jupyter-widgets/ipywidgets/pull/3390
- https://github.com/jupyter-widgets/ipywidgets/pull/3393

- Introductions
    - Gayle Ollington
      * [Jupyter Community Building Committee Charter](https://jupyter.org/governance/communitybuildingcommittee.html)
      * [December 2021 Jupyter Community Update](https://blog.jupyter.org/jupyter-community-2021-update-84c5cd3c5e75)
      * Keep an eye out for an upcoming blog post with a survey about JupyterCon
    - Steve Purves
      - co-founder of [CurveNote](https://curvenote.com/) 
      - Rowan works a lot on Myst, Steve works a lot on Thebe
      - Working on ipywidgets in Thebe
- Presentations on ipywidgets implementations
  - Pete Blois next week
  - Advertise so others who are implementing ipywidgets can attend
  - Steve Purves: widgets in curvenote
- Echo state updates backwards compatible: https://github.com/jupyter-widgets/ipywidgets/pull/3394
  - Vidar is reviewing this in the next 36 hours or so.
  - Look at https://github.com/jupyter-widgets/ipywidgets/issues/3111 - is Maarten's case where frontend and backend go out of sync taken care of?
- 7.x and 8.x releases
  - Jason and Itay can do an RC release of 7.7 and 8.0 after #3394 gets merged.
- Jupyter Books (Steve)
  - Executable book implementing something like Like Brett Victor's [Tangle](http://worrydream.com/Tangle/) library
    - Jason once implemented a ipywidgets widget that used something like the Tangle library to do widgets in text.
  - Reference also recent discussions on Discourse about interpolating variable values into Markdown
  - SageCell
    - https://sagecell.sagemath.org/
    - https://utmost.aimath.org/ (see also the PreText books that integrate Sage Cell, like )
- Executable docs for ipywidgets using JupyterLite: https://twitter.com/martinRenou/status/1498321869363154947?s=20&t=stPowTab4Z9CE00CBPDbrg

**Comment by jtpio** [comment]
## 8 March 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Steve Purves | Curvenote | @stevejpurves |
| Xiao Chen | Databricks | @xiaochen-db |
|tonyfast|Quansight|@tonyfast|
|William Stein|CoCalc| @williamstein|
|Jared Thompson|CCC|@afonit|
| Pete Blois | Google | @blois |
|Florian Wetschoreck | Databricks | @FlorianWetschoreck |
|Tobias Krabel | Databricks | @tkrabel |
| Martin Renou | QuantStack | @martinRenou |
|Hossein Falaki| Databricks|@falaki|
|Michael Piatek| Databricks|@michaelp-db|
|Itay Dafna| Bloomberg| @ibdafna|
|Trung Le|QuantStack|@trungleduc|


- Introductions


### Merged PRs:
- [Echo state updates in a backwards-compatible way](https://github.com/jupyter-widgets/ipywidgets/pull/3394)
- [Limit dependabot to run-time dependencies only](https://github.com/jupyter-widgets/ipywidgets/pull/3402)

### PRs to review:
- [Backport echo_update messages to 7.x](https://github.com/jupyter-widgets/ipywidgets/pull/3400)
- [Add migration guide for 8.0](https://github.com/jupyter-widgets/ipywidgets/pull/3405)
- [Follow up on echo updates](https://github.com/jupyter-widgets/ipywidgets/pull/3407)
- [Make sure we serialize dictionary into valid JSON #3408](https://github.com/jupyter-widgets/ipywidgets/pull/3408)

### 7.7 and 8.0 RC/release plans
- RC this week?

### Other items

- Recording meetings




### [Pete Blois](https://github.com/blois): ipwyidgets on Colab

- [Notes](https://colab.research.google.com/gist/blois/9c9c66b4e1e9672b123f2ed8cda9091d/colab-widgets.ipynb#scrollTo=6JEQUuO2DBGI)

**Comment by jtpio** [comment]
## 15 March 2022
| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
|William Stein|CoCalc|@williamstein|


- William demo'd CoCalc RTC Markdown
- Releases! 7.7.0rc0, 8.0.0rc0
- Will release 7.7.0 final today
- Remaining 8.0 issues on Milestone: https://github.com/jupyter-widgets/ipywidgets/milestone/30
    - Migration guide: https://github.com/jupyter-widgets/ipywidgets/pull/3405
- New issue/questions, needs triage: 
    - https://github.com/jupyter-widgets/ipywidgets/issues/3411
        - Is `on_click` an observer?
    - https://github.com/jupyter-widgets/ipywidgets/issues/3412


- Question from a user (william) -- comments about upgrading from this -- probably not too bad.  Critical to be aware of backend version; make sure it is at least 7.
  - https://github.com/sagemathinc/cocalc/issues/5784
```
~/cocalc/src/packages/frontend$ grep widget package.json 
    "@jupyter-widgets/base": "^1.2.5",
    "@jupyter-widgets/controls": "^1.5.3",
    "@jupyter-widgets/output": "^1.1.5",
    "@phosphor/widgets": "^1.9.3",
```

- William: I can quickly demo ipywidgets in a whiteboard, since I have it working now...

**Comment by jtpio** [comment]
## 29 March 2022

No meeting.

**Comment by jtpio** [comment]
## 5 April 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Itay Dafna | Bloomberg | @ibdafna |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
||||
||||

- Remaining 8.0 issues on Milestone: https://github.com/jupyter-widgets/ipywidgets/milestone/30


- Scipy tutorial was accepted! Scheduled Tuesday morning 8am-12 noon.

What other libraries should we try to convert?

- ipyleaflet
- pythreejs

8.0 release concerns
- 8.0rc is not working in vs code. Hypothesis: if the user has 8.0 installed in the kernel, why doesn't the 7.0 javascript work. It used to work with the 8.0 beta.
- https://github.com/microsoft/vscode-jupyter/issues/8552



April 19 - Don presenting on widgets in vs code

**Comment by jtpio** [comment]
## 12 April 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Itay Dafna | Bloomberg LP | @ibdafna |



Issues over last week:
- https://github.com/jupyter-widgets/ipywidgets/issues/3433 - leading to creation of https://github.com/b3m2a1/ActiveHTMLWidget (example at https://github.com/jupyter-widgets/ipywidgets/issues/3433#issuecomment-1095669732)

Items:
- Conda-forge releases of RC0
- public path discussion: https://github.com/jupyter-widgets/widget-ts-cookiecutter/blob/master/%7B%7Bcookiecutter.github_project_name%7D%7D/src/extension.ts and https://github.com/jupyter-widgets/widget-cookiecutter/blob/969471848da747d5aa562b8ab00988e3117e4ccf/%7B%7Bcookiecutter.github_project_name%7D%7D/js/lib/extension.js#L8 (originating in https://github.com/jupyter-widgets/widget-cookiecutter/pull/38)
  - Can we use webpack's ['auto' public path](https://webpack.js.org/guides/public-path/#automatic-publicpath) instead of hardcoding a public path? I think this allows us to consolidate the nbextension AMD module with the dist/ CDN AMD module
- Conferences people are going to?
  - Scipy: @ibdafna, possibly @jasongrout
    -  Jupyter Widgets tutorial
  - PyCon US: possibly @jasongrout
- Interesting library for doing EDA workflows: https://pypi.org/project/dtale/. Does not seem to use ipywidgets.

Next week Don Jayamanne is presenting on widgets in VS Code. Please advertise.
- Gitter
- Discourse
- JLab dev meeting

**Comment by jtpio** [comment]
## 19 April 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Frederic Collonval | QuantStack | @fcollonval |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Martin Renou | QuantStack | @martinRenou |


Issues over the last week:

Items:
- Frontends migrating from widgets 7 to widgets 8, supporting both: https://github.com/jupyter-widgets/ipywidgets/issues/3429
  - How common will it be that a frontend has to support core widgets from ipywidgets 7 and 8 simultaneously?
  - Two extensions:
    - plugin for Jupyterlab ipywidgets 8 widget manager, and plugin that registers the ipywidgets 8 core controls
    - plugin that registers the ipywidgets 7 core controls
- Migration guide https://github.com/jupyter-widgets/ipywidgets/pull/3405
- ipyleaflet and jupyterlab-sidecar migrated: [jupyterlab-sidecar](https://github.com/jupyter-widgets/ipywidgets/pull/3405#issuecomment-1100566870) and [ipyleaflet](https://github.com/jupyter-widgets/ipywidgets/pull/3405#issuecomment-1100576009)
- More migrations:
  - Jason do ipympl (Ian added tests for ipywidgets 8? https://github.com/matplotlib/ipympl/pull/461)
  - Vidar do pythreejs
- ipywidgets 8 final release delayed a bit


- JSON spec


- Don's presentation postponed one week, will be next Tuesday

**Comment by jtpio** [comment]
## 26 April 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Martin Renou | QuantStack | @martinRenou |
| Itay Dafna | Bloomberg | @ibdafna |
| Layne Sadler | AIQC | @aiqc |
|William Stein|CoCalc/SageMath|@williamstein|
| Florian Wetschoreck | Databricks | @FlorianWetschoreck |
| Tobias Krabel | Databricks | @tkrabel |
| Frederic Collonval | QuantStack | @fcollonval |

Issues
- Frontends migrating from widgets 7 to widgets 8
  - VS Code - waiting until release and customer requests
  - Colab - actually on 8 prerelease already - user code is built against 7, but run against 8
    - patched the lumino change
    - 
  - CoCalc - may be difficult, due to having one single frontend codebase, but definitely having old versions of ipywidgets installed in kernels "forever", since we support old images.  "Maybe I can just somehow change the comm messages to be compatible?";  Pete actually does basically this in Colab.  Jason says this will mostly work, but there are some subtle differences, and they will publish an official spec change doc.
- jupyter.widget.control error causing confusion: https://github.com/jupyter-widgets/ipywidgets/issues/2257
  - This is caused when ipywidgets hasn't been imported yet
  - We should move forward on the comm JEP
- migration guide, ready for another review https://github.com/jupyter-widgets/ipywidgets/pull/3405
  - did the phosphor shim
  - added note about backbone extend
- https://github.com/jupyter-widgets/ipywidgets/pull/3442
- https://github.com/jupyter-widgets/ipywidgets/issues/3443
- Cookiecutter updated so that it works with ipywidgets 7 and 8 https://github.com/jupyter-widgets/widget-ts-cookiecutter/pull/115

ipywidgets in vscode-jupyter:

https://github.com/microsoft/vscode-jupyter/wiki/Component:-IPyWidgets

Virtualization:
- Colab: uses intersection observer to defer creation of the iframes until they have been scrolled into view but never un-renders them when they go out of view. We write an 'output_height' attribute in the cell's metadata to specify the placeholder height on notebook load. Overall this sounded easier than it was in practice- thought it would take a few days, ended up taking maybe 2 months to stabilize (welcome to production)
- VS Code: We do the same in VS Code as well, render only when scrolled into view and never un-render them.
- Cocalc: is now fully virtual (except iframe html) https://github.com/sagemathinc/cocalc/pull/5844 using https://virtuoso.dev/ and seems to work.  Motivation for me is very complicated notebooks where having content rendered in the dom at all can be expensive when working with the notebook.  (Other approach is to make this less expensive -- maybe things are significantly extra expensive in cocalc.)

**Comment by jtpio** [comment]
## 03 May 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Itay Dafna | Bloomberg | @ibdafna |


- William Stein presenting on ipywidgets in Cocalc on 17 May.
- Remaining tasks for 8.0 release. Timeline.
- Itay to add notes to the migration guide about `backbone.js` version update between ipywidgets 7.x and 8.x, which affects some `IClassicComm` mock implementations.

**Comment by jtpio** [comment]
## 10 May 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Jason Grout | Databricks | @jasongrout |
| Itay Dafna | Bloomberg | @ibdafna |



- William Stein presenting on ipywidgets in Cocalc on 17 May.
  - Jason will advertise on Gitter, Discourse, JLab meeting
- @DonJayamanne's talk about how Jupyter Widgets are implemented in VS Code is posted at https://www.youtube.com/watch?v=C821QIYcxd0. Thanks again Don!
- PyCon
  - PyScript and Pyodide
  - Cap for memory usage of pyodide is 2g
  - Pythreejs maintainership
- npm does not pull in 6.0.0-rc: 
    - `^1 || ^2 || ^3 || ^4 || ^5 || ^6.0.0-rc.0` will install 4.1.0, while `^1 || ^2 || ^3 || ^5 || ^6.0.0-rc.0` will install 6.0.0-rc.0
      - Use the semver calculator: https://semver.npmjs.com/
    - next vs latest tag?
    - date of publish of different versions?

- Remaining work for 8.0
    - https://github.com/jupyter-widgets/ipywidgets/pull/3442
    - https://github.com/jupyter-widgets/ipywidgets/issues/3439
    - Merged [#3405](https://github.com/jupyter-widgets/ipywidgets/pull/3405), please add more things as you see them.
    - Potential final release next week
- [Itay] Scipy tutorial
  - Iterating on content
  - Meeting with organizers to finalize content, etc.
  - `react-ipywidgets`: https://github.com/widgetti/react-ipywidgets

**Comment by jtpio** [comment]
## 17 May 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Pete Blois | Google | @blois |
| William Stein | CoCalc | @williamstein |
| Blaec Bejarano | CoCalc | |



- es6 vs backbone extend
  - Currently ipywidgets 7.x and 8.x are generating es6 classes (not backbone.extend)
  - https://github.com/googlecolab/colab-cdn-widget-manager/blob/main/src/swizzle.ts - bridges between backbone extend and es6 classes. Just wrap the class being exposed with swizzle() and it can be constructed via ES6 new or backbone's constructor.
  - k3d uses backbone.extend
  - Pythreejs also uses backbone.extend
  - Vidar made a note in the migration guide about migrating from backbone.extend


- Remaining work for 8.0
    - https://github.com/jupyter-widgets/ipywidgets/pull/3455 - blocker for 8.x - documenting spec differences
    - [Add trimmed down readme file](https://github.com/jupyter-widgets/ipywidgets/pull/3461)
    - Merged [#3405](https://github.com/jupyter-widgets/ipywidgets/pull/3405), please add more things as you see them.
    - Potential final release next week
    - Vidar investigating the es6 migration

- Long-term API
  - Lumino is a dependency
    - Lifecycle methods - can we prefer native methods that external lifecycle events can call
  - jupyter-services is a dependency
  - font-awesome - William wrote a jquery plugin that simulates a font-awesome and manually adds the font-awesome look-alikes icons in cocalc
  - typeset - depends on mathjax? - should be a method on the manager
  - assumptions about DOM and CSS
  - Theming - should widgets be using jupyterlab variables directly?
    - Could define only jupyter widgets variables, then have the jlab manager define those



Third-party widgets
- ipycytoscape
- pythreejs
- ipyleaflet
- cadquery
- ipyvuetify
- ipymaterialui
- ipycanvas
- bqplot
- plotly
- nglviewer
- ipydatagrid
- k3d
- ipyslickgrid
- qgrid
- ipyvolume
- ipysheet

See https://www.npmjs.com/browse/depended/@jupyter-widgets/base for hundreds more.

Python to JS translater: https://github.com/sagemathinc/jsage (JPython, inside of JSage, is BSD 2-clause)

---

- William Stein presenting on ipywidgets in CoCalc
  - https://cocalc.com/wstein/support/ipwidgets-in-cocalc

**Comment by jtpio** [comment]
## 24 May 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| William Stein | CoCalc | @williamstein |
| Itay Dafna | Bloomberg | @ibdafna |





- [x] William: I implemented binary buffers for cocalc's ipywidgets frontend, and now k3d works pretty well at scale.   But I have a  question for devs I got stuck on: namely -- how does the widget manager know that all the comm messages needed to create the view have been received?  How does it predict the future?  This stumps me.
  - Messages are serialized over the iopub channel. Each message should be processed fully before later messages are processed
- Pruning the widget tree for persistence
  - Maarten Breddels implemented pruning at some point
  - Python version here: https://github.com/jupyter-widgets/ipywidgets/blob/master/python/ipywidgets/ipywidgets/embed.py

- ipywidgets 8.0
  - Several PRs updating webpack infrastructure: https://github.com/jupyter-widgets/ipywidgets/pull/3465, https://github.com/jupyter-widgets/ipywidgets/pull/3464
  - current diff of model specs between 7 and 8 (`git diff 7.x -- packages/schema/jupyterwidgetmodels.latest.md`): https://gist.github.com/jasongrout/07d52b85aaffe994db0e2a0c018a6c78
    - Why do border and overflow adopt different conventions between the overarching value vs the individual values?
    - GH best attempt: https://github.com/jupyter-widgets/ipywidgets/compare/7.x...master#diff-76c58c499131f42cddfc4b58686947ab4bc1c76e3e291ab06516109e9a9c1ce3
    - [ ] Update the schema header for ipywidgets 8.x
    - [ ] Summarize changes in the release notes
  - [ ] 


Continuing discussion from last time about post 8.0:

- font-awesome - William wrote a jquery plugin that simulates a font-awesome and manually adds the font-awesome look-alikes icons in cocalc
  - We could explicitly support, say 100 icons that cover most icons third party widgets are currently using, and the manager is responsible for providing those icons in a way that is consistent with the platform. This makes it easier for widget implementors to support the icons without having to explicitly use fontawesome. (william: I **really** like this proposal.)
- typeset - depends on mathjax? - should be a method on the manager
  - Add it as a minor release in the base manager interface


---

- William: another quick question -- what about a list of jupyter clients that support widgets?  E.g., I was recently testing and learned that datalore does now.
  - https://www.jetbrains.com/dataspell/
  - datalore.jetbrains.com
- [ ] Add a table to the docs with links to frontends that implement ipywidgets?
   - do in such a way that is in no way an endorsement
   - put it on a wiki.
   - another table of third-party ipywidgets

- Vidar: porting pythreejs to ipywidgets 7 and 8
- Jason: looking at updating docs with diff, look at todo items
- William: porting k3d to cocalc

Community workshop proposal for ipywidgets:
- October 2022 or Jan 2023
- Paris, London, San Francisco, Berlin

**Comment by jtpio** [comment]
## 31 May 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| William Stein (until 10am PDT due to https://sites.google.com/view/vantageseminar, slides: https://math.mit.edu/~drew/vantage/BhargavaSlides.pdf) | CoCalc | @williamstein |
| Itay Dafna | Bloomberg | @ibdafna |

- PRs/issues for 8.0: https://github.com/jupyter-widgets/ipywidgets/milestone/30

- I would be interested in a very quick discussion of https://github.com/jupyter-widgets/ipywidgets/pull/3114 since something similar is at the top of my personal todo list for cocalc.  (I want this more for https://cocalc.com/share, but it's the same problem.)
  - on roadmap?: definitely not, due to this being jupyter classic, and we are moving to jupyterlab, which already solves this issue, for the most part.
  - what about nbconvert / publishing
     - nbconvert is aware of widgets; save widget state.
     - e.g., this is used by our docs.  displayed using html manager
  - what if the widget state is BIG (say 50MB), e.g., 3d graphics and buffers?  2d plots + bqplot -- see this all the time.  blows up size of notebook.
     - we originally had all notebooks saved every widgets, and this was not optimal due to extensive interactive use and no garbage collection, so notebooks would rapidly increase in size
     - then we made it off by default.
     - make it very explicit when you want to save this way.
     - related discussion: sidecar files.  jupytext does this.
  - and generally what scares you about this PR?  If anything?
  - Maarten has "Solara"...
  - Subtle definition of "live widget".   View counter.
  - But in cocalc since it garbage collects if not displayed anywhere.
  - Browser memory and file size are both big concerns.
  - What about screenshot?    Widget's responsibility.  Or serialize dom...     
  - [william] Thanks!

- ipydatagrid ipywidgets 8.0 upgrade: https://github.com/bloomberg/ipydatagrid/pull/282

**Comment by jtpio** [comment]
## 7 June 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Itay Dafna | Bloomberg | @ibdafna |


- PRs/issues for 8.0: https://github.com/jupyter-widgets/ipywidgets/milestone/30
- Community Workshop: 
- Tutorial: 
  07/12/2022, 8:00am-12:00pm CDT, The Jupyter Interactive Widget Ecosystem, Matthew Craig, Itay Dafna, Mariana Meireles, Youness Bennani, Ian Hunt-Isaak
- Can we have a widgets 8 release by June 17, which is when the tutorial instructions are due?
  - I think so!
- Databricks supports ipywidgets! https://docs.databricks.com/notebooks/ipywidgets.html
- Borders vs overflow attributes: in order to have the possibility to adjust borders independently, we went with border_left/right/top/bottom, with convenience properties to set the border. For the overflow attribute, we went with our normal convention of preferring the single shorthand CSS notation over individual properties.

**Comment by jtpio** [comment]
## 14 June 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Matt Craig | Minnesota State University Moorhead | @mwcraig |
| Martin Renou | QuantStack | @martinRenou |
| William Stein | SageMath/CoCalc | @williamstein |
| Itay Dafna | Bloomberg | @ibdafna |


- update wiki page: https://github.com/jupyter/jupyter/wiki/Jupyter-Widgets
  - Add "Documentation" column to the widget table
  - Add ipywidgets to the text above, then let the table be sorted
- Release [7.7.1](https://github.com/jupyter-widgets/ipywidgets/issues/3483)
- Question about `titles` on `Tab` and `Accordion` -- is this a list or a tuple?
- Status of several packages with respect to version 8 support:
    + [`ipyleaflet`](https://github.com/jupyter-widgets/ipyleaflet/pull/968) -- open PR that allows ipywidgets 8 only (see js dependencies).
        * [ ] Will this support `ipywidgets` `7` **and** `8`, or just `8`?
    + [`bqplot`](https://github.com/bqplot/bqplot/pull/1404) -- Open PR but it is out-of-date: 
        * [x] Will this support `ipywidgets` `7` **and** `8`, or just `8`? Both 7 and 8.
        * [ ] Should use version `6` instead of `5` for `@jupyter-widgets/base`, see [this line](https://github.com/bqplot/bqplot/pull/1404/files#diff-e51a40ac250c9696142466f114f754161e7e5102c0cdb5354548b757deb272f6R73).
        * [ ] Should use `5` instead of `4` for `@jupyter-widgets/controls`
        * [ ] Currently [uses a custom scales package](https://github.com/bqplot/bqplot/pull/1404/files#r818653111) -- this PR needs to merger first, I'm guessing: https://github.com/bqplot/bqscales/pull/49
    + [`bqscales`](https://github.com/bqplot/bqscales/pull/49) -- a dependency of `bqplot`. Some fixes needed:
        * [x] Will this support `ipywidgets` `7` **and** `8`, or just `8`? Both 7 and 8
        * [ ] Should use version `6` instead of `5` for `@jupyter-widgets/base`, see [this line](https://github.com/bqplot/bqplot/pull/1404/files#diff-e51a40ac250c9696142466f114f754161e7e5102c0cdb5354548b757deb272f6R73).
    + [`ipympl`](https://github.com/matplotlib/ipympl/pull/461) -- Open PR but may be incomplete:
        * [ ] It looks like this still imports `phosphor` instead of `lumino`.
    + [`ipytree`](https://github.com/QuantStack/ipytree) -- No PR yet.
        * [ ] Follow the migration guide to open a PR.
    + [`ipycanvas`](https://github.com/martinRenou/ipycanvas) -- needs a pull request.
        * [ ] Follow the migration guide to open a PR.
    + [`sidecar`](https://github.com/jupyter-widgets/jupyterlab-sidecar/pull/86) -- PR is open.
        * [ ] Per [this comment](https://github.com/jupyter-widgets/jupyterlab-sidecar/pull/86#issuecomment-1100566571) one commit needs to be reverted before merging.
    + [`pythreejs`](https://github.com/jupyter-widgets/pythreejs/pull/378) -- PR is open, but
        * [x] Will this support `ipywidgets` `7` **and** `8`, or just `8`? Just 8
        * [x] Need to remove `rc1` from `package.json` before merging.
    + [`ipycytoscape`](https://github.com/cytoscape/ipycytoscape/blob/master/package.json) -- 🎉 works! 🎉
        * [ ] Not sure why, though, see https://github.com/cytoscape/ipycytoscape/blob/master/package.json#L60 which doesn't include the  latest `@jupyter-widgets/base`
            * Expected, notebook and lab serve up whatever they have.
    + [`ipyevents`](https://github.com/mwcraig/ipyevents) -- 🎉 works! 🎉
        * [ ] Not sure why, though, see https://github.com/mwcraig/ipyevents/blob/main/package.json#L47 which doesn't include the  latest `@jupyter-widgets/base`
* https://github.com/jupyter-widgets/ipyleaflet/pull/988 - this change takes out the hardcoded public paths, would love to propagate to other widget packages
- border vs overflow attribute changes result from last time
- The new comm echo was causing an issue in newer versions of Lab (execution state would go wonky). Will be fixed in next release of Lab: https://github.com/jupyterlab/jupyterlab/issues/12616

### Additional discussion:

- Documentation
  - Make interactive examples with JupyterLite. Use the JupyterLite-sphinx extension
  - https://github.com/phoebe-project?language=jupyter+notebook
  - http://linear.ups.edu/fcla/section-DM.html
  - https://sagecell.sagemath.org/
  - https://github.com/executablebooks/thebe
  - https://jupyterlite-sphinx.readthedocs.io/en/latest/
  - [william] what about [a "Gallery" like matplotlib has](https://matplotlib.org/stable/gallery/index.html) but for ipywidgets?  Like we have for interact in sage: https://wiki.sagemath.org/interact (these are of course actually using ipywidgets)
  - https://alanmbarr.com/blog/four-types-software-documentation/
    - https://doc.sagemath.org/ is like this, IMHO...
  - https://dash.plotly.com/layout
  - Lacking a gallery
  - Qt reference docs are very detailed: https://doc.qt.io/qt-6/qtreewidget.html
  - Change "low level tutorial" to "low level explanation"
  - Include tutorial video in landing page
  -

**Comment by jtpio** [comment]
## 21 June 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Matt Craig | Minnesota State University Moorhead | @mwcraig |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| William Stein | SageMath/CoCalc | @williamstein |

- Tutorial: use jupyterlite?
  - See https://github.com/jupyter-widgets/ipywidgets/pull/3491 (and the rendered docs at https://ipywidgets--3491.org.readthedocs.build/en/3491/)

- 7.7.1: Jason to release in the next day or two
- 8.0: Coordinate over Gitter, and check in about possibly releasing Friday
- PyData London sprint:

**Comment by jtpio** [comment]
## 28 June 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| William Stein | SageMath/CoCalc | @williamstein |
| Itay Dafna | Bloomberg | @ibdafna |


- 8.0:
    - User migration docs 7.x->8.0 https://github.com/jupyter-widgets/ipywidgets/pull/3501

- william: for what it is worth, I'm working on finally implementing custom messages from the browser to the kernel here https://github.com/sagemathinc/cocalc/pull/6011  Also buffers: https://github.com/sagemathinc/cocalc/issues/6022

**Comment by jtpio** [comment]
## 5 July 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Matt Craig | Minnesota State University Moorhead | @mwcraig |

Meeting canceled

**Comment by jtpio** [comment]
## 12 July 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| William Stein | SageMath/CoCalc | @williamstein |

- Changes since last meeting:
    - https://github.com/jupyter-widgets/ipywidgets/pull/3501 - User migration docs merged.
    - https://github.com/jupyter-widgets/ipywidgets/pull/3508 - webpack public path for AMD modules improvements
    - https://github.com/jupyter-widgets/ipywidgets/pull/3510 - Backwards compatability for description_tooltip via deprecation

Remaining tasks 8.0:
- https://github.com/jupyter-widgets/ipywidgets/issues/3455 - Make a diff of the spec that is readable on RTD
- https://github.com/jupyter-widgets/ipywidgets/issues/3429 - Migration plan for frontends



(william) Community question: 
- thoughts about https://github.com/jupyter-widgets/ipywidgets/issues/3516  "I am not sure how this could be solved unless we drastically change ipywidgets' architecture."
  - discussed in the jupyter server meeting
  - ipywidgets uses comms, but comms exist outside ipywidgets, and have unknown usage that can't be completely replaced by rtc. 

- JupyterLite discussion: cool new features for docs.

**Comment by jtpio** [comment]
## 19 July 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| William Stein | SageMath/CoCalc | @williamstein |
|Matt Craig|Minnesota State University Moorhead | @mwcraig |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Itay Dafna | Bloomberg | @ibdafna |

- Report on Scipy and the tutorial there
  - Went really well
    - Twice we had some serious technical issues with connecting computers, etc., but attendees probably didn't even notice.
    - Not much changed from last year:
      - JupyterLite
      - Widgets 8
      - Building an application
  - Probably about 20 people there, so way down from most years
  - Jupyter is still the big thing
  - Papyri talk by Matthias was excellent
  - VegaFusion by Jon Mease is an ipywidget!
  - JDAViz (scsci-provided platform for web images) is widget-based. Based on glue, etc.
  - No widget sprint, Matt worked on bqplot and other things, though. There was a little Jupyter sprinting as well.

- [8.0 milestone](https://github.com/jupyter-widgets/ipywidgets/milestone/30)
    + Went over open PRs. Some decisions:
        - Will add this to docs: https://github.com/jupyter-widgets/ipywidgets/issues/3257
        - Other decisions made in the individual issues/PRs

- [Automatically set public path](https://github.com/jupyter-widgets/widget-cookiecutter/pull/103)

**Comment by jtpio** [comment]
## 26 July 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| William Stein | SageMath/CoCalc | @williamstein |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Matt Craig | Minnesota State University Moorhead | @mwcraig |


- ipywidgets 8
    - [3524](https://github.com/jupyter-widgets/ipywidgets/pull/3524): experiment is concluded, close the pr and watch for issues in the community after release
    - [3528](https://github.com/jupyter-widgets/ipywidgets/pull/3528) - minor tweak, then merge
    - [3455](https://github.com/jupyter-widgets/ipywidgets/issues/3455) - is this issue good enough for frontend authors? Answer: yes, but add a link to the changelog

**Comment by jtpio** [comment]
## 2 August 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Itay Dafna | Bloomberg LP | @ibdafna |
| William Stein | SageMath/CoCalc | @williamstein |
| Jason Grout | Databricks | @jasongrout |
| Martin Renou | QuantStack | martinRenou | 



- ipywidgets 8
    - [#3528](https://github.com/jupyter-widgets/ipywidgets/pull/3528) - I made one tweak, but William still had a request for a change.
- Jeremy is working on a Voila PR to convert to ipywidgets 8 and a jlab based framework https://github.com/voila-dashboards/voila/pull/846. Also moving ipyvolume to ipywidgets 8 https://github.com/widgetti/ipyvolume/pull/411

- Martin
  - moving dependency of ipykernel to the new comm package
  - https://github.com/martinRenou/comm
  - https://github.com/ipython/ipykernel/pull/973
  - https://github.com/jupyter-widgets/ipywidgets/pull/3533
  - https://github.com/jupyter-xeus/xeus-python/pull/548
  - Is this backwards compatible, at least so that we can release it with 8.1?
    - We can remove the ipykernel dependency in 8.1

**Comment by jtpio** [comment]
## 9 August 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Jason Grout | Databricks | @jasongrout |
| A. T. Darian | QuantStack | @afshin |
| Matt Craig | Minnesota State University Moorhead | @mwcraig |
| Itay Dafna | Bloomberg LP | @ibdafna |

- ipywidgets 8
  - ipywidgets 8 testing
    - Want positive confirmation about ipywidgets 8 working in various environments
      - Jason tested JupyterLab and basic html manager embedding
      - Things to test:
          - [ ] Test HTML manager in 3rd party widget docs and nbconvert (https://github.com/jupyter-widgets/ipywidgets/pull/3536) - OutputWidget, styling
          - [x] Test ipywidgets with JupyterLab (including old version, 1, 2, 3.0.0)
  - PRs needing review:
    - https://github.com/jupyter-widgets/ipywidgets/pull/3539
      - [ ] Vidar to add commits linking changelog and migration guide and adding, for example, file upload migration material to migration guide
    - https://github.com/jupyter-widgets/widget-cookiecutter/pull/87
  - [ ] Blog post: Vidar will draft a paragraph for extension authors about you no longer going to need to update 3rd party libs every time lab has a major release. After the next release :)
- Lumino 2
  - [Request for help](https://github.com/jupyterlab/team-compass/issues/158) (`jupyterlab/team-compass`)
  - [Lumino 2 project board](https://github.com/orgs/jupyterlab/projects/4/views/1)
  - JupyterLab widget manager may need a major upgrade? It will certainly need code changes to deal with the JLab 4 API changes
  - We'll have to decide between tradeoffs:
    - bump base by a minor version, with the caveat that .luminoWidget is either Lumino 1 or 2, with the idea that most people will be compatible
    - bump base by a major version, and libraries that are okay with either lumino 1 or 2 just make the dependency an or (||).
- Widget workshop
  - ideally 3-5 days?
  -

**Comment by jtpio** [comment]
## 16 August 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Jason Grout | Databricks | @jasongrout |
| A. T. Darian | QuantStack | @afshin |
| Sylvain Corlay | QuantStack | @sylvaincorlay |


- ipywidgets 8
  - https://github.com/jupyter-widgets/ipywidgets/pull/3540
  - https://github.com/jupyter-widgets/ipywidgets/pull/3541
  - New doc PR:
    - Removes the building custom widget guide from the index until it can be updated to use the js cookiecutter (or ts cookie cutter updated)
    - Updating the widget author migration guide:
      - point to js cookiecutter
      - update language around 8.0
      - add a note about the public path update in the js cookiecutter
  - Blog post
    - Jason can work on it late this week or early next week. Any help will speed this up, of course :).
- [Native iterators in Lumino 2](https://github.com/jupyterlab/lumino/pull/346)

**Comment by jtpio** [comment]
## 23 August 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Martin Renou | QuantStack | @martinRenou |
| Pete Blois | Google | @blois|
| Maarten Breddels| | @maartenbreddels|
| Jeremy Tuloup | QuantStack | @jtpio |



* ipywidgets 8.0
  * Quick recap of release
    * [Done] Check to make sure there is a cap on the dependencies for ipywidgets 8
  * 

TODO issues:
* Mappings as selection options: mutually exclusive decisions:
  * https://github.com/jupyter-widgets/ipywidgets/pull/3556
  * https://github.com/jupyter-widgets/ipywidgets/pull/3557
    * Should `Dropdown(options={'a': 1, 'b': 2})` work? It worked in 7.x, we removed it in 8.x, but you can use `Dropdown(options={'a': 1, 'b': 2}.items())`. It causes people's code to break - should we reintroduce being able to pass in a dict?
      * Removed because dict used to be unordered
      * Removed also because the ownership model is unclear - if you change the dict, the options don't change.
  * Decision: Merge https://github.com/jupyter-widgets/ipywidgets/pull/3557 to reverse the removal

* What to allow through sanitizer?
    * https://github.com/jupyter-widgets/ipywidgets/pull/3565
      * Allow 'class', but make sure there is a high bar for future sanitizer additions, and ideally a standardized way for how to consider such requests. Compare to lab? Some better "gold standard"?

* Handle `KernelWidgetManager` in the JupyterLab `OutputModel`
    * https://github.com/jupyter-widgets/ipywidgets/pull/3561
    * Probably ok 

* Are these kind of typings changes backwards compatible? Fixes?
  * https://github.com/jupyter-widgets/ipywidgets/pull/3554

* Restore `.widgets` ?:
  * https://github.com/jupyter-widgets/ipywidgets/issues/3562
      * PR: https://github.com/jupyter-widgets/ipywidgets/pull/3122
      * [Possible fix](https://github.com/widgetti/react-ipywidgets/commit/6fb07845d143f02165c11405ed5bec20391b09c5)
      * Should we have a deprecated .widgets and .widget_types on Widget for backwards compatibility?
        * Yes, let's make the upgrade easier with a deprecated getter
        * [ ] Maarten doing a PR

* TODO issues: https://github.com/jupyter-widgets/ipywidgets/milestone/30
* precommit: generally positive

**Comment by jtpio** [comment]
## 30 August 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Pete Blois | Google | @blois |
| Itay Dafna | Bloomberg LP | @ibdafna |


* 8.0.x milestone for 8.0 compat issues
  * Went through 8.0.x milestone
* widget community workshop: *"The future of ipywidgets"*
  * London, proposed date 17th October, 4 days, can be shifted
    * More ideas:
      * ipywidgets in iframes
      * A clear comm protocol (like Colab?) https://github.com/googlecolab/colabtools/blob/main/packages/outputframe/lib/index.d.ts
      * Some people may want to participate remotely
      * 
* controls version number: https://github.com/jupyter-widgets/ipywidgets/issues/3560

**Comment by jtpio** [comment]
## 06 Sep 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Pete Blois | Google | @blois |

* [8.0.x milestone](https://github.com/jupyter-widgets/ipywidgets/milestone/37)
    * https://github.com/jupyter-widgets/ipywidgets/pull/3567 What is next?
        * To fully resolve issue, I think it would need a note in the migration guide?
    * https://github.com/jupyter-widgets/ipywidgets/pull/3569 - Maybe a unit test?.
* Blog post? Last item on [8.0.0 milestone](https://github.com/jupyter-widgets/ipywidgets/issues?q=is%3Aopen+is%3Aissue+milestone%3A8.0)
  * Jason make a medium post for editing and dressing up
  * Also post on discourse
* Pete's proposal:

Here's the public/stable API that Colab exposes to outputs:
https://github.com/googlecolab/colabtools/blob/main/packages/outputframe/lib/index.d.ts

* It hangs off of window which I know doesn't work well for JupyterLab.
* The API is intended to be bare-bones. A single typings file, no dependencies on any libraries or frameworks.
* The API is intended to be as strongly typed as possible, to attempt to avoid vagaries that can lead to breaking changes.
* Our implementation uses JS scopes to hide private members from the objects exposed, to prevent use of unintentionally exposed members. This is for compatibility, it's not considered a security boundary.

Discussion of an improved `%%javascript` magic and I mention how an API like Colab's could be exposed in a portable manner that would work in JupyterLab as well.
https://github.com/ipython/ipython/issues/13376#issuecomment-989040562

Bokeh is a good example of what we see elsewhere:
https://github.com/bokeh/bokeh/blob/0d0f40b7b0be373c2a58a2b99241de98ea49dba2/src/bokeh/core/_templates/autoload_nb_js.js#L121

They look for window.Jupyter and if present attempt to load classic notebook implementation modules via requirejs.

Where is this discussion most appropriate?
Pete did an implementation of this JS api in a Jlab extension: https://github.com/blois/js-module-renderer

**Comment by jtpio** [comment]
## 13 Sep 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Vidar T Fauske | JP Morgan Chase | @vidartf |
| Jason Grout | Databricks | @jasongrout |


* Adding a [trove classifier](https://packaging.python.org/en/latest/specifications/core-metadata/#classifier-multiple-use) for Jupyter widgets
  * Similar to adding the PR adding JupyterLab classifiers [here](https://github.com/jupyterlab/jupyterlab/issues/9538)
  * [List of current classifiers](https://pypi.org/classifiers/)
  * Perhaps these classifiers?
    * `Framework :: Jupyter :: Jupyter Widgets :: 7`
    * `Framework :: Jupyter :: Jupyter Widgets :: 8`
    * Or should it be singular `Jupyter Widget`? (Jason thinks plural). [Examples](https://pypi.org/classifiers/) of each:
      * Plural
        * JupyterLab :: Extensions :: Themes
        * JupyterLab :: Extensions :: Mimerenderers
        * Topic :: Text Processing :: Filters
        * Topic :: Text Editors :: Word Processors
        * Topic :: Terminals
        * Topic :: Software Development :: Libraries
        * Topic :: Software Development :: Assemblers
        * Topic :: Multimedia :: Sound/Audio :: Mixers
        * Topic :: Desktop Environment :: Window Managers
      * Singular
        * Programming Language :: Python :: Implementation :: CPython
        * Programming Language :: APL
        * Operating System :: BeOS

* ipywidgets 8.0.x: https://github.com/jupyter-widgets/ipywidgets/milestone/37
  * Jason still to merge changes in https://github.com/jupyter-widgets/ipywidgets/pull/3569
* blog post
  * Jason to put up a draft of the blog post about 8.0
* 2-factor auth turned on for jupyter-widgets github org
* workshop
* Jupyter Widgets Council
  * October 3 target date
* VS Code supporting ipywidgets 8
  * https://github.com/microsoft/vscode-jupyter/issues/11217
  * Previous exploration at https://github.com/jupyter-widgets/ipywidgets/pull/3524

**Comment by jtpio** [comment]
## 20 Sep 2022

| Name | Affiliation | GitHub |
|------|-------------|--------|
| Jason Grout | Databricks | @jasongrout |
| Itay Dafna | Bloomberg LP | @ibdafna |
||||
||||


- 8.0.x 

Blog post draft: https://blog.jupyter.org/ipywidgets-8-0-df1f22ef8278
